src/lib/db/migrations/20230329093154_transaction-table-modify-cols.ts

```diff
@@ -0,0 +1,14 @@
+import { Knex } from 'knex'
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.table('transaction', function (table) {
+    table.dropColumn('token_id')
+    table.dropColumn('token_type')
+    table.enu('api_type', ['match2', 'order', 'capacity'], { useNative: true, enumName: 'api_type' })
+    table.enu('transaction_type', ['creation', 'proposal', 'accept'], { useNative: true, enumName: 'transaction_type' })
+  })
+}
+
+export async function down(knex: Knex): Promise<void> {
+  await knex.schema.dropTable('transaction')
```

Comment:
the down function should be a reverse of `up`to put the tablel back in the state it was in before e.g. https://github.com/digicatapult/inteli-api/blob/main/migrations/20220526113916_fix_tx_status_col.js

Reply:
you can readd `token_type` with 
```
    def
      .enum('token_type', ['DEMAND', 'MATCH2'], {
        enumName: 'type',
        useNative: true,
      })
      .notNullable()
```
but then you need to set a value for every row because it's `notNullable`. Which I think would mean looking at each rows `api_type` and mapping `match2` -> `MATCH2` and `order || capacity` -> `DEMAND`. But that means in the `up` you also need to do the reverse of `token_type` -> `api_type` for each row. And how do you decide if a `DEMAND` is an `order` or `capacity`.

It's complicated and also pointless anyway because we don't have any databases deployed with reall data